cmd/clusterctl/cmd/version_checker_test.go

```diff
@@ -123,6 +123,58 @@ New clusterctl version available: v0.3.8 -> v0.3.9-alpha.0
 https://github.com/foo/bar/releases/v0.3.9-alpha.0
 `,
 		},
+		{
+			name: "returns message if cli version is a pre-release of the latest available release",
+			cliVersion: func() version.Info {
+				return version.Info{
+					GitVersion: "v0.3.9-alpha.0",
+				}
+			},
+			githubResponse: `{"tag_name": "v0.3.9", "html_url": "https://github.com/foo/bar/releases/v0.3.9"}`,
+			expectErr:      false,
+			expectedOutput: `
+New clusterctl version available: v0.3.9-alpha.0 -> v0.3.9
+https://github.com/foo/bar/releases/v0.3.9
+`,
```

Comment:
If we ever had our own official pre-releases, I wanted clusterctl to be able to notify the user via the output.